doc/developer/design/20220516_transactional_consistency.md

```diff
@@ -0,0 +1,456 @@
+implementation details on the read holds).
+
+There is an additional restriction that objects on different timelines cannot currently be read in the same transaction.
+Details on this are out of scope for this document but more details can be found
+here: [https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/20210408_timelines.md](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/20210408_timelines.md)
+
+##### Write Only Transactions
+
+Write only transactions allow a client to perform one or more writes atomically. They currently only allow writes to a
+single table. They also only allow `INSERT` statements (no `DELETE`, `UPDATE`, `CREATE`
+, `INSERT INTO SELECT`, etc). All writes within a transaction are buffered in a session local variable and are not
+visible to any other session. When the transaction commits, a timestamp is selected by the coordinator and all writes
+are sent to STORAGE with the selected timestamp. Commits do not return an OK message to the client until all writes have
+been made durable.
+
+NOTE: This is written with the assumption that `append` STORAGE commands for a single object are idempotent, atomic, and
+don’t return until they are durable. While this isn’t currently the case in Materialize, it is being actively worked on
+and should be completed soon.
+
+#### Single Read/Write Statements
+
+- `INSERT` statements behave identically to write-only transactions containing a single statement.
+- `SELECT` statements behave very similarly to read-only transactions containing a single statement. A timestamp is
+  selected for the read, the read is performed at that timestamp, and the results are sent back to the client. However,
+  no read holds are taken and the timestamp is selected only using the objects in the query (not all objects in the
+  schema like read-only transactions).
+- `COPY TO` statements perform a read and then writes the results to an external source. Currently, we only
+  support `COPY TO STDOUT` which behaves identically to a normal `SELECT`. In the future if we add more destinations
+  then the read will be executed exactly as described in the `SELECT` bullet above, and we can exclude the write from
+  this discussion because it is to an external system.
+- `COPY FROM` statements read from an external source and then writes the results to a table. Currently, we only
+  support `COPY FROM STDIN` which behaves identically to a normal `INSERT`. In the future if we add more sources then
+  the the write will be executed exactly as described in the `INSERT` bullet above, and we can exclude the read from
+  this discussion of consistency because it is from an external source.
+
+#### DDL
+
+NOTE: Sinks are excluded from the discussion in this section.
+
+- `CREATE` statements create new objects such as tables, sources, and views. They have five steps
```

Comment:
tw nit: never write out the number of elements in a list. Instead rely on some infinitive verb, or some trailing preposition.

> `CREATE` statements create new objects such as tables, sources, and views by:

Note that the expectation is that the verb tense of the bullets then "completes" the leading sentence.

Reply:
Thanks, fixed